sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2841,6 +2860,265 @@ class DatasetSuite extends QueryTest
     checkDataset(Seq(seqMutableSet).toDS(), seqMutableSet)
     checkDataset(Seq(mapMutableSet).toDS(), mapMutableSet)
   }
+
+  // below tests are related to SPARK-49960 and TransformingEncoder usage
+  test("Incorrect derived nullability with TransformingEncoder - non nullable") {
```

Comment:
I am a bit confused by the phrasing here. Did we fix the issue? Or are you showing it is broken?

Reply:
rephrased into the positive